lib/galaxy/webapps/galaxy/api/jobs.py

```diff
@@ -155,7 +155,8 @@ def show(self, trans, id, **kwd):
                 job_stderr=job.job_stderr,
                 stderr=job.stderr,
                 stdout=job.stdout,
-                job_messages=job.job_messages
+                job_messages=job.job_messages,
+                job_remap=job.remappable()
```

Comment:
```suggestion
```
remappable used to be something that is a bit of an internal hack and I don't think this should be part of this API endpoint (it only is part of the somewhat internal `build_for_rerun`). It answers 2 different questions (are there dependent jobs ? did the job produce any dataset collection instances) that I think should be dealt with on the client or that should at least be 2 keys in the response.
Looking closer at this I think `if self.output_dataset_collection_instances:` isn't even correct, you'd probably need to ask if this job has a `ImplicitCollectionJobAssociation`.

Maybe we can take this out for now and look at a replacement in a separate PR ?

Reply:
The rest is great, makes sense to move the logic to the model.

Reply:
> Maybe we can take this out for now and look at a replacement in a separate PR ?

fine for me